ompi/mca/coll/base/coll_base_alltoall.c

```diff
@@ -235,15 +235,90 @@ int ompi_coll_base_alltoall_intra_pairwise(const void *sbuf, int scount,
     return err;
 }
 
-
-int ompi_coll_base_alltoall_intra_bruck(const void *sbuf, int scount,
-                                         struct ompi_datatype_t *sdtype,
-                                         void* rbuf, int rcount,
-                                         struct ompi_datatype_t *rdtype,
-                                         struct ompi_communicator_t *comm,
-                                         mca_coll_base_module_t *module)
+/*
+ * 
+ * Function: ompi_coll_base_alltoall_intra_k_bruck using O(logk(N)) steps
+ * Accepts:  Same arguments as MPI_Alltoall
+ * Returns: MPI_SUCCESS or error code
+ * 
+ * Description: This method extends ompi_coll_base_alltoall_intra_bruck to handle any
+ * radix k(k >= 2).
+ * 
+ *   Example on 6 nodes with k = 4
```

Comment:
`s/nodes/ranks/`

Reply:
replace nodes with nodes/ranks?

Reply:
Yes, the terminology is ranks

Reply:
sorry, I was using `sed` shorthand.  I was requesting you change the comment from `Example on 6 nodes` to `Example on 6 ranks` because nodes implies distinct hosts participating, but this algorithm only considers how many ranks are involved.